tests/integration/article.test.js

```diff
@@ -224,60 +225,330 @@ describe("# Articles endpoints", () => {
+              .send({ email: "fake@email.com", password: "243hjgudsgdgh" })
+              .end((err, res) => {
+                if (!err) fakeToken = res.body.token;
+                done(err || undefined);
+              });
+          }
+        });
+    });
+
+    describe("Updating a specific article", () => {
+      it("should add the new tags on an article", done => {
+        chai
+          .request(app)
+          .put(`/api/v1/articles/${validSlug}`)
+          .set("Authorization", `Bearer ${token}`)
+          .send({ tagsList: ["politics", "music", "love"] })
+          .end((err, res) => {
+            expect(res.status).equals(CREATED);
+            expect(res.body.message).to.contain("Updated");
+            expect(res.body)
+              .to.haveOwnProperty("article")
+              .to.be.an("object");
+            done();
+          });
       });
+
+      it("should throw the error if the current user is not the author of that article", done => {
+        chai
+          .request(app)
+          .put(`/api/v1/articles/${validSlug}`)
+          .set("Authorization", `Bearer ${fakeToken}`)
+          .end((err, res) => {
+            expect(res.status).equals(UNAUTHORIZED);
+            expect(res.body.message).to.contain(
+              "You can only update the article you authored"
+            );
+            done();
+          });
+      });
+
+      it("should throw the bad request error if the description is invalid", done => {
+        chai
+          .request(app)
+          .put(`/api/v1/articles/${validSlug}`)
+          .set("Authorization", `Bearer ${token}`)
+          .send({ description: "Love" })
+          .end((err, res) => {
+            expect(res.status).equals(BAD_REQUEST);
+            expect(res.body.message).to.contain(
+              '"description" length must be at least 10 characters long'
+            );
+            done();
+          });
+      });
+    });
+
+    describe("Deleting a specific article", () => {
+      it("should throw the error if the current user is not the author of that article", done => {
+        chai
+          .request(app)
+          .delete(`/api/v1/articles/${validSlug}`)
+          .set("Authorization", `Bearer ${fakeToken}`)
+          .end((err, res) => {
+            expect(res.status).equals(UNAUTHORIZED);
+            expect(res.body.message).to.contain(
+              "You can only delete the article you authored"
+            );
+            done();
+          });
+      });
+
+      it(`should delete the article with slug ${validSlug}`, done => {
+        chai
+          .request(app)
+          .delete(`/api/v1/articles/${validSlug}`)
+          .set("Authorization", `Bearer ${token}`)
+          .end((err, res) => {
+            expect(res.status).equals(OK);
+            expect(res.body.message).to.contain("Deleted");
+            done();
+          });
+      });
+
+      it("should throw the error if article is not found", done => {
```

Comment:
Which error code is more suitable for this case?
I think 404 is suitable because the ressource is not found..